sources/Context.cpp

```diff
@@ -17,11 +17,26 @@
 
 #include "Context.h"
 
+#include "Log.h"
+
 #include <tuple>
 
 namespace dfl {
-Context::Context(const std::string& networkFilepath) : networkManager_(networkFilepath) {
-  networkManager_.onNode(std::bind(&Context::processSlackNode, this, std::placeholders::_1));
+Context::Context(const std::string& networkFilepath, const std::string& configFilepath) :
+    networkManager_(networkFilepath),
+    config_(configFilepath),
+    slack_node_{} {
+  auto automatic_slack_node = networkManager_.getSlackNode();
+  if (automatic_slack_node.is_initialized() && !config_.isAutomaticSlackBusOn()) {
+    slack_node_ = *automatic_slack_node;
+  } else {
+    // automatic slack node not given: we must compute it ourselves
+    if (!automatic_slack_node.is_initialized() && config_.isAutomaticSlackBusOn()) {
```

Comment:
Maybe change the name "automatic_slack_node" to "found_slack_node" ?

Reply:
Done